app/controllers/sessions_controller.rb

```diff
@@ -11,27 +11,27 @@ def new
 
   def github
     return show_login_restriction unless role_id = github_authorization.role_id
-    login(role_id: role_id)
+    login(role_id: custom_role_or_default(role_id))
   end
 
   def google
     return show_login_restriction unless allowed_to_login
-    login(role_id: Role::VIEWER.id)
+    login(role_id: custom_role_or_default(Role::VIEWER.id))
   end
 
   def ldap
     return show_login_restriction unless allowed_to_login
-    login(role_id: Role::VIEWER.id)
+    login(role_id: custom_role_or_default(Role::VIEWER.id))
   end
 
   def gitlab
     return show_login_restriction unless allowed_to_login
-    login(role_id: Role::VIEWER.id)
+    login(role_id: custom_role_or_default(Role::VIEWER.id))
```

Comment:
should login take care of custom_role_or_default ?

Reply:
I was on the fence about that... I'll make the change 👍

Reply:
Actually, now I think about it I think it makes sense to leave this logic in each action, in case it ever needs to change for a specific authentication method?